### ISSUE TITLE
Requirement to include markers for blank and omitted pages

### ISSUE THREAD

**Issue by mattgarrish** [w3c/epub-specs#1502]
If an author omits some content from the digital publication, or if we allow blank pages to be skipped, should the page list have gaps for the missing content or should we require all the source pages be listed?

For example, we could use `a` elements without an `href` attribute to represent omitted pages:

```
<ol>
   ...
   <li><a href="c02.html#page43">43</a></li>
   <li><a>44</a></li>
   <li><a href="c03.html#page45">45</a></li>
   ...
</ol>
```

**Comment by gregoriopellegrino** [comment]
I don't have a strong opinion, my only doubt is: if we enter this requirement, how can we automatically check it? I don't think it's 100% possible, so it would require manual checking.

**Comment by mattgarrish** [comment]
> how can we automatically check it

It'll never be fully automatable, but a lot of the accessibility requirements aren't. 

You can check for gaps at the start and between numbers, though, so for most simple texts the only check would be the last page number. Appendices and other oddly numbered pages will be the most problematic.

**Comment by mattgarrish** [comment]
It might also be possible to naively check for errors by requiring a total page count in the metadata. If the sum of the list items in the page list doesn't equal the total page count, then something is amiss.

**Comment by dauwhe** [comment]
Do we know what commonly-used tools do now? And what publishers do? I know there are a lot of page lists out there, and I'd be wary of new requirements that might require a lot of work to update existing tools and content.

**Comment by mattgarrish** [comment]
> And what publishers do?

This gets to the heart of the problem, namely that we find variance in what people produce. #1503 poses a similar question.

We need more detailed expectations for page breaks and the page list, otherwise even the most useless of page lists gets a pass (i.e., an empty page list technically passes our current requirements as we don't say anything about how complete the list has to be).

I only put the example in only to show a complete list is not out of the realm of possibility to do. The primary question we need to answer is the completeness one itself, both for markers and the page list.

**Comment by TzviyaSiegman** [comment]
I think this is more a question of what it looks like to a user. If I am relying in the page numbers being read aloud as i go through a book, and it skips from 43 to 45, that is pretty confusing. I am not sure that anyone uses the numbering that way, but as a publisher that has been my concern with skipping or re-sequencing. I would love to hear from users more than publishers.

**Comment by GeorgeKerscher** [comment]
Very good point. Because there has traditionally been a problem with the completeness of content, knowing that a page is blank is great to know. So, when I go from page 43, encounter a blank page 44 and move on to page 45,  life is good. If I go to 43 and then encounter 45, I say to myself, crap, this title is missing pages.

**Comment by avneeshsingh** [comment]
Predictability helps people with accessibility needs. Therefore as a visually impaired user, I dislike missing pages.

**Comment by mattgarrish** [comment]
We'll need to figure out how we harmonize the variations we're likely to encounter, though.

As it stands, page breaks are optional. That means not only do you not have to provide them, but even if you provide them you don't have to provide a complete set. You could skip blank pages or you could even skip content pages. All cases pass. Do we require all page breaks be identified when they are included, or is it only a recommendation?

If we require completeness, we'll need to figure out what that means for omitted content. Does the publisher have to insert page breaks that say the content has been omitted, or is it only a recommendation?

Are we preventing publishers from rearranging content from print to digital if that content is on a numbered page, as it will cause readers to encounter strange jumps in numbering, or is it only a recommendation?

I was trying to break these issues up to deal with them separately, but if it makes sense to tackle them all at once that's fine with me. My gut feeling is we'll end up with recommendations (or even best practices), but still not fail deviations from the recommendations.

**Comment by avneeshsingh** [comment]
There are publications in which static pages map to a hard copy or a fixed layout copy. On the other hand there are publications which are created independent of hard copies or fixed layout versions.
Conceptually thinking: 
1. if the metadata declares print equivalent, then we should have all the pages.
2. But if metadata does not indicate it then publisher has flexibility.

Looking from EPUB accessibility perspective, we can have no. 1 in specs or in techniques. Techniques may be more safe, but would like to know the opinions for it.

**Comment by clapierre** [comment]
>if the metadata declares print equivalent, then we should have all the pages.

I think we need to have something then in our 1.1 spec regarding this:

https://w3c.github.io/epub-specs/epub33/a11y/#sec-page-nav
Currently we have:
A conformant EPUB Publication has to meet the following criteria when it includes page navigation:

It MUST provide a means of locating the page break locations.
It MAY include page break markers.
It MUST identify the source of the page breaks.

Would we add 

"It SHOULD include all pages from the source including any blank pages"

**Comment by mattgarrish** [comment]
(I've renamed this issue as I don't think we need to concern ourselves with the page list beyond whatever resolution we come to in #1503.)

I understand the appeal of wanting all pages represented, but is this realistic for publishers? Back when I did daisy production, I wouldn't have questioned this since we just chopped the spine and reproduced every page in order. But that's not my understanding of how publishers create digital versions from their print.

I don't think we can consider blank pages in isolation from omitted content. If we're going to require or recommend all pages be accounted for, we have to be ready for pushing publishers to include page markers for the content they dropped. That also means we'll have to accept that users may sometimes have to navigate a series of pages that just repeat they're omitted content. Are we sure this is ideal?

If it was only blank pages we had to be concerned about, a recommendation might not be that problematic. Perhaps we might want to think of this in terms of best practice?

**Comment by mattgarrish** [comment]
I wonder if this can be made to match what I proposed for the page list, too, so:

- Level A  - no requirement for markers - matches our current requirement
- Level AA - markers must be included in reflowable publications for all content reproduced - not requiring blank or omitted content
- Level AAA - markers must be included in reflowable publications for all pages from the source whether blank or omitted

**Comment by mattgarrish** [comment]
The one problematic aspect of both the page list and markers is that they're currently conditional based on how the publication is produced or where it is expected to be distributed.

That leaves a big loophole in terms of enforcing that I'm not sure what we can do about (i.e., it will be hard to ever make these formal wcag requirements unless we solve this more reliably).

**Comment by iherman** [comment]
The issue was discussed in a [meeting](https://www.w3.org/publishing/groups/epub-wg/Meetings/Minutes/2021-03-11-epub-a11y#section4) on 2021-03-11

- no resolutions were taken

<details><summary><i>View the transcript</i></summary>

### 4. pages related issues 

_See github issue [#1503](https://github.com/w3c/epub-specs/issues/1503), [#1502](https://github.com/w3c/epub-specs/issues/1502), [#1501](https://github.com/w3c/epub-specs/issues/1501), [#1500](https://github.com/w3c/epub-specs/issues/1500), [#1542](https://github.com/w3c/epub-specs/issues/1542)._



**Matt Garrish:**: Updated EPUB accessibility for addressing a part of the following static pages issues. Some requirements can be addressed by EPUB Accessibility while others should go to best practices.

> *Avneesh Singh:* [https://w3c.github.io/epub-specs/epub33/a11y/index.html#sec-page-nav](https://w3c.github.io/epub-specs/epub33/a11y/index.html#sec-page-nav)

**Matt Garrish:** we had different questions about pagelists  
… one propose is to structure the requirements for pagelists  
… e.g. level A: some pages in the pagelist  
… level AA: all content pages linked in pagelist  
… level AAA: all pages (no exclusion, even blank pages)  
… I didn't move over with my requirements  

**Avneesh Singh:** I don't think these issues can be added in the guidelines, maybe the best place is best practices  

**Tzviya Siegman:** we had similar in PDFs with "this page intentionally left blank", where we left blank pages for starting the chapters on page right  
… I think that marking blank pages can be really confusing  
… maybe we can use ARIA  

**Avneesh Singh:** yes, but then someone will say that ARIA is not only for AT  

**Charles LaPierre:** I'm not sure if Read Aloud by the Reading system can get that information  

**Matt Garrish:** for sure we can investigate the use of ARIA label or something similar  

**Avneesh Singh:** I think this task force should focus on the requirements Matt put in the issues  
… we can go through the issue tracker  
… and discuss there  

**Matt Garrish:** we have a issue about the ordering of the pagelist  
… if I have the pages moved in the digital version, what happen to the pagelist?  

**Tzviya Siegman:** I have several examples of this  

**Avneesh Singh:** I think for AT users it would be useful to have an alert  

**Charles LaPierre:** maybe we can put in in the accessibility summary (metadata), but do we need to have some requirements for reading systems?  

**Gregorio Pellegrino:** maybe something like play order attribute in NCX tocs  

---


</details>

**Comment by mattgarrish** [comment]
As discussed, closing this with the changes in https://github.com/w3c/epub-specs/pull/1558
